lib/improver/tests/cube_combiner/test_CubeCombiner.py

```diff
@@ -98,25 +99,360 @@ class Test__repr__(IrisTest):
+        cube = create_cube_with_threshold()
+        changes = 'delete'
+        result = plugin.update_coord(cube, 'threshold', changes)
+        found_key = 'threshold' in [coord.name() for coord in result.coords()]
+        self.assertArrayEqual(found_key,
+                              False)
+
+    def test_coords_deleted_fails(self):
+        """Test update_coord fails to delete coord of len > 1. """
+        plugin = CubeCombiner('-')
+        cube = create_cube_with_threshold()
+        changes = 'delete'
+        msg = "Can only remove a coordinate of length 1"
+        with self.assertRaisesRegexp(ValueError, msg):
+            plugin.update_coord(cube, 'time', changes)
+
+    def test_warning_messages_with_delete(self):
+        """Test warning message is raised correctly when deleting coord. """
+        plugin = CubeCombiner('-', warnings_on=True)
+        coord_name = 'threshold'
+        cube = create_cube_with_threshold()
+        changes = 'delete'
+        warning_msg = "Deleted coordinate"
+        with warnings.catch_warnings(record=True) as warning_list:
+            warnings.simplefilter("always")
+            plugin.update_coord(cube, coord_name, changes)
+            self.assertTrue(any(item.category == UserWarning
+                                for item in warning_list))
+            self.assertTrue(any(warning_msg in str(item)
+                                for item in warning_list))
+
+    def test_coords_update_fail_points(self):
+        """Test that update_coord fails if points do not match. """
+        plugin = CubeCombiner('-')
+        cube = create_cube_with_threshold()
+        changes = {'points': [2.0, 3.0]}
+        msg = "Mismatch in points in existing coord and updated metadata"
+        with self.assertRaisesRegexp(ValueError, msg):
+            plugin.update_coord(cube, 'threshold', changes)
+
+    def test_coords_update_fail_bounds(self):
+        """Test update_coord fails if shape of new bounds do not match. """
+        plugin = CubeCombiner('-')
+        cube = create_cube_with_threshold(threshold_values=[2.0, 3.0])
+        changes = {'bounds': [0.1, 2.0]}
+        msg = "The shape of the bounds array should be"
+        with self.assertRaisesRegexp(ValueError, msg):
+            plugin.update_coord(cube, 'threshold', changes)
+
+    def test_coords_update_bounds_succeed(self):
+        """Test that update_coord succeeds if bounds do match """
+        plugin = CubeCombiner('-')
+        cube = create_cube_with_threshold(threshold_values=[2.0, 3.0])
+        cube.coord('threshold').guess_bounds()
+        changes = {'bounds': [[0.1, 2.0], [2.0, 3.0]]}
+        result = plugin.update_coord(cube, 'threshold', changes)
+        self.assertArrayEqual(result.coord('threshold').bounds,
+                              np.array([[0.1, 2.0], [2.0, 3.0]]))
+
+    def test_coords_update_fails_bounds_differ(self):
+        """Test that update_coord succeeds if bounds do match """
```

Comment:
I think this docstring isn't quite right.

Reply:
Changed